gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/impl/sql/conn/GenericAuthorizer.java

```diff
@@ -153,12 +153,13 @@ public void authorize( Activation activation, int operation) throws StandardExce
                 checkAccess();
                 int sqlAllowed = RoutineAliasInfo.NO_SQL;
                 StatementContext ctx = this.lcc.getStatementContext();
-                if (ctx != null) {
-                  sqlAllowed = ctx.getSQLAllowed();
-                } else if (perms != null && perms.size() > 0) {
+                if (perms != null && perms.size() > 0) {
                   // direct call from GfxdSystemProcedures.authorizeColumnTableScan
                   sqlAllowed = RoutineAliasInfo.READS_SQL_DATA;
                 }
+                else if (ctx != null) {
```

Comment:
Will suggest to also initialize ctx rather than before the perms check: else if ((ctx = this.lcc.getStatementContext()) != null) {